### PR TITLE
Use MinimalScene by default

### PR DESCRIPTION
**Pull request by chapulina**
# 🎉 New feature

Part of https://github.com/ignitionrobotics/ign-gui/issues/137

## Summary

Ideally we'd deprecate `GzScene3D` on Fortress so we stop maintaining 2 scenes. For that to happen, we need to achieve feature parity with `MinimalScene`. We need the following, and maybe more:

* https://github.com/ignitionrobotics/ign-gazebo/pull/900
* https://github.com/ignitionrobotics/ign-gazebo/pull/983
* https://github.com/ignitionrobotics/ign-gazebo/pull/988

~~It would also be good to get this in, so it's easier to configure the GUI and users don't have to repeat lots of plugins for the 3D scene:~~ Going without it for now

* https://github.com/ignitionrobotics/ign-gazebo/pull/1012

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    Q_OBJECT
+
+    /// \brief Constructor
+    /// \param[in] _parent Parent item
+    public: explicit EntityContextMenuItem(QQuickItem *_parent = nullptr);
+
+    /// \brief Set the entity context menu hanlder
+    /// \param[in] _EntityContextMenuHanlder Entity context menu hanlder
+    public: void SetEntityContextMenuHanlder(
+      EntityContextMenuHanlder &_EntityContextMenuHanlder);
```
Comment (chapulina) [inline src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh:91]:
```suggestion
      const EntityContextMenuHanlder &_entityContextMenuHandler);
```

File: src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  {
+    Q_OBJECT
+
+    /// \brief Constructor
+    /// \param[in] _parent Parent item
+    public: explicit EntityContextMenuItem(QQuickItem *_parent = nullptr);
+
+    /// \brief Set the entity context menu hanlder
+    /// \param[in] _EntityContextMenuHanlder Entity context menu hanlder
+    public: void SetEntityContextMenuHanlder(
```
Comment (chapulina) [inline src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh:90]:
```suggestion
    public: void SetEntityContextMenuHandler(
```

File: src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.cc [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    {
+      auto cam = std::dynamic_pointer_cast<rendering::Camera>(
+        this->scene->NodeByIndex(i));
+      if (cam)
+      {
+        if (std::get<bool>(cam->UserData("user-camera")))
+        {
+          this->camera = cam;
+
+          igndbg << "Spawn plugin is using camera ["
```
Comment (chapulina) [inline src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.cc:82]:
```suggestion
          igndbg << "Entity context menu plugin is using camera ["
```

File: src/gui/gui.config [hunk 4]
```diff
@@ -38,10 +38,77 @@
   <scene>scene</scene>
   <ambient_light>0.4 0.4 0.4</ambient_light>
   <background_color>0.8 0.8 0.8</background_color>
-  <camera_pose>6 0 6 0 0.5 3.14</camera_pose>
+  <camera_pose>-6 0 6 0 0.5 0</camera_pose>
 </plugin>
 
-<!-- Play / pause / step -->
+<!-- Plugins that add functionality to the scene -->
+<plugin filename="GzSceneManager" name="Scene Manager">
```
Comment (chapulina) [inline src/gui/gui.config:45]: We need to add `EntityContextMenuPlugin` here

File: src/gui/plugins/drop_from_fuel/DropFromFuel.hh [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace ignition
+{
+namespace gazebo
+{
+  class DropFromFuelPrivate;
+
+  /// \brief Allows to DropFromFuel models and lights using te gui event
+  /// DropFromFuelFromDescription
+  class DropFromFuel : public ignition::gui::Plugin
```
[the remark this answers is not in the document]
Comment (chapulina) [reply src/gui/plugins/drop_from_fuel/DropFromFuel.hh]: Done in 0874cd0

File: src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+#include <ignition/rendering/Camera.hh>
+
+namespace ignition
+{
+namespace gazebo
+{
+  class EntityContextMenuPrivate;
+
+  /// \brief This plugin is in charge to show the entity context menu when the
```
Comment (jennuine) [inline src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh:35]: nit
```suggestion
  /// \brief This plugin is in charge of showing the entity context menu when the
```

File: src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  class EntityContextMenuItem : public QQuickItem
+  {
+    Q_OBJECT
+
+    /// \brief Constructor
+    /// \param[in] _parent Parent item
+    public: explicit EntityContextMenuItem(QQuickItem *_parent = nullptr);
+
+    /// \brief Set the entity context menu hanlder
+    /// \param[in] _entityContextMenuHandler Entity context menu hanlder
```
Comment (jennuine) [inline src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh:89]:
```suggestion
    /// \param[in] _entityContextMenuHandler Entity context menu handler
```

File: Migration.md [hunk 8]
```diff
@@ -47,6 +47,24 @@ since pose information is being logged in the `changed_state` topic.
...
+  order to get the same functionality as `GzScene3D`, users need to add the
+  following plugins:
+      + `MinimalScene`: base rendering functionality
+      + `GzSceneManager`: adds / removes / moves entities in the scene
+      + `EntityContextMenuPlugin`: right-click menu
+      + `InteractiveViewControl`: orbit controls
+      + `CameraTracking`: Move to, follow, set camera pose
+      + `MarkerManager`: Enables the use of markers
+      + `SelectEntities`: Select entities clicking on the scene
+      + `Spawn`: Functionality to sawn entities into the scene via GUI
```
Comment (jennuine) [inline Migration.md:60]:
```suggestion
      + `Spawn`: Functionality to spawn entities into the scene via GUI
```

**Review by jennuine — APPROVED** [review]
LGTM :tada: with some minor nits

File: src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.cc [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+
+/////////////////////////////////////////////////
+void EntityContextMenuPrivate::OnRender()
+{
+  if (nullptr == this->scene)
+  {
+    this->scene = rendering::sceneFromFirstRenderEngine();
+    if (nullptr == this->scene)
+    {
+      return;
```
[the remark this answers is not in the document]
Comment (jennuine) [reply src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.cc:67]: Nope, scene may be created later

File: src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.cc [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+#include "EntityContextMenuPlugin.hh"
+
+#include <memory>
+#include <utility>
+
+#include <QtQml>
+
+#include <ignition/common/Console.hh>
+#include <ignition/common/MouseEvent.hh>
```
[the remark this answers is not in the document]
Comment (jennuine) [reply src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.cc]: 55d9790d62086645bcd8fa42be4bea5f32d948ad

File: src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh (hunk 6 above)
Comment (jennuine) [reply src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh]: 55d9790d62086645bcd8fa42be4bea5f32d948ad

File: src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh (hunk 7 above)
Comment (jennuine) [reply src/gui/plugins/entity_context_menu/EntityContextMenuPlugin.hh]: Found and fixed more `hanlders` 55d9790d62086645bcd8fa42be4bea5f32d948ad

File: Migration.md (hunk 8 above)
Comment (jennuine) [reply Migration.md]: 55d9790d62086645bcd8fa42be4bea5f32d948ad

**Pull request merged**